File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int letters = 0;    
int words = 1;     
int sentences = 0;
string text;

void count(int n);
void result(int index);

int main(void)
{
    letters = 0;    //Set counter variables to zero
    words = 1;      // Word count starts at 1
    sentences = 0;
    text = get_string("Text: ");     // Request user input for text

    int n = strlen(text);                   // Gets stringlength
    
    count(n);
    
    float L = ((float)letters / (float)words) * 100;
    float S = ((float)sentences / (float)words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);   // Apply formula

    result(index);
}




void count(n)
{
    for (int i = 0; i < n; i++)
    {
    
        if ((text[i] > 64 && text[i] < 91) || (text[i] > 96 && text[i] < 123))
        {
            letters ++;         // Counts letters
        }
    
        if (text[i] == 32)
        {
            words ++;           // Counts words
        }
    
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences ++;       // Counts sentences
        }

    }
}


void result(index)
{
    if (index < 1)                      // Print results
    {
        printf("Before Grade 1\n");
    }

    else if (index > 1 && index < 16)
    {
        printf("Grade %i\n", index);
    }

    else
    {
        printf("Grade 16+\n");
    }
    
}

File: pset6/readability/readability.py
from cs50 import get_string


def main():
    
    text = get_string("Text: ")
    text = text.lower()
    
    print(text)
    words, ltrs, sents = count(text)

    L = (ltrs / words) * 100
    S = (sents / words) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)
    
    result(index)


def count(n):
    
    words = 1
    ltrs = 0
    sents = 0
    
    for i in n:
        
        if i.isalpha() == True:
            
            ltrs += 1
            
        if i == " ":

            words += 1
            
        if i in ["!","?","."]:
            
            sents += 1
            
    return words, ltrs, sents

def result(y):
    
    if y < 1:
        
        print("Before Grade 1")
        
    elif y > 1 and y < 16:
        
        print(f"Grade {y}")
        
    else:
        
        print("Grade 16+")
        
 
main()